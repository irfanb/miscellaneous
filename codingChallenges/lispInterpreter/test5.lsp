(defun fib (n)
  (if (< n 2)
      n
      (+ (fib (- n 1))
         (fib (- n 2)))))

(defun fact (n)
  (if (<= n 1)
    1
    (* n (fact (- n 1)))))
