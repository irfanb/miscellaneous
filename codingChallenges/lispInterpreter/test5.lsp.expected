found BEGINEXPRESSION (
found id defun
found id fib
found BEGINEXPRESSION (
found id n
found ENDEXPRESSION )
found BEGINEXPRESSION (
found id if
found BEGINEXPRESSION (
found LESSTHAN
found id n
found EXPRESSIONELEMENT 2
found ENDEXPRESSION )
found id n
found BEGINEXPRESSION (
found PLUS
found BEGINEXPRESSION (
found id fib
found BEGINEXPRESSION (
found MINUS
found id n
found EXPRESSIONELEMENT 1
found ENDEXPRESSION )
found ENDEXPRESSION )
found BEGINEXPRESSION (
found id fib
found BEGINEXPRESSION (
found MINUS
found id n
found EXPRESSIONELEMENT 2
found ENDEXPRESSION )
found ENDEXPRESSION )
found ENDEXPRESSION )
found ENDEXPRESSION )
found ENDEXPRESSION )
found BEGINEXPRESSION (
found id defun
found id fact
found BEGINEXPRESSION (
found id n
found ENDEXPRESSION )
found BEGINEXPRESSION (
found id if
found BEGINEXPRESSION (
found LEQ
found id n
found EXPRESSIONELEMENT 1
found ENDEXPRESSION )
found EXPRESSIONELEMENT 1
found BEGINEXPRESSION (
found TIMES
found id n
found BEGINEXPRESSION (
found id fact
found BEGINEXPRESSION (
found MINUS
found id n
found EXPRESSIONELEMENT 1
found ENDEXPRESSION )
found ENDEXPRESSION )
found ENDEXPRESSION )
found ENDEXPRESSION )
found ENDEXPRESSION )
